### PR TITLE
Add a `&null` option to `from-lines` and `to-lines`

### LINKED ISSUE #1070: `to-lines` should support a configurable EOL char
I was working on a change to the elvish unit test framework to avoid deadlocks due to the shell writing more output than can be buffered by a pipe. So I wrote a test that did, among other things, this:

```
repeat (* 128 1024) x | str:join '' | to-lines
```

What I want is exactly 128 KiB of the letter "x", without a trailing newline, written to stdout. What I get as a consequence of the `to-lines` is the 128 KiB of "x"s plus a newline. The solution was to replace the above with this:

```
repeat (* 128 1024) x | each [c]{ print $c }
```

I'm perfectly happy with the solution above. What I'm wondering is how many other people would have reached for the `to-lines` command like I did and perhaps not notice they were getting an unwanted newline? Also, should there be an alternative, such as `to-lines &n`, to suppress the newline that might be easier to discover and more intuitive for some people?

### PR DESCRIPTION
**Pull request by krader1961**
This change, combined with other changes to `edit:command-history`,
makes feeding its output into `fzf` extremely fast. Which makes it a
practical alternative to using the builtin Ctrl-R binding for searching
command history. It's also just generally useful to have efficient ways
to process "lines" that are null terminated rather than newline (or
cr-nl on Windows).

Resolves #1070
Related #1053

### CODE REVIEW

**Comment by xiaq** [comment]
I added a comment on #1070 on how I prefer this to be implemented.

**Comment by krader1961** [comment]
@xiaq: This change now implements `from-terminated` and `to-terminated` commands rather than a `&null` option by `{from to}-lines`.

**Review by xiaq — APPROVED** [review]
Thanks for the PR. I'll merge this and apply some minor fixups after merging.

File: pkg/eval/builtin_fn_io.go
Comment (krader1961) [reply pkg/eval/builtin_fn_io.go:767]: Those flags are quite deliberate. The `-e` only marks newlines. It's presence helps ensure that if `to-terminated` unexpectedly includes a newline that is visible. The `-v` is redundant when `-t` is used but I'm pretty sure that in the past I've used `cat` commands where that was not true. Better safe than sorry.

File: pkg/strutil/chop.go
Comment (krader1961) [reply pkg/strutil/chop.go:3]: Huh, I thought it was supported. Oh well.

File: pkg/eval/builtin_fn_io.go
Comment (xiaq) [reply pkg/eval/builtin_fn_io.go:767]: Oh, I didn't realize that `-e` is needed to make the absence of newline more obvious, so it is actually needed.

> The `-v` is redundant when `-t` is used but I'm pretty sure that in the past I've used `cat` commands where that was not true. Better safe than sorry.

Probably SunOS/Solaris? https://illumos.org/man/1/cat

POSIX only documents a single option, `-u`, which probably means that there is platform variety in even the behavior of `-v`. Best to change this to use `slurp` instead.

**Pull request merged**
